policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceAsyncProxyTest.java

```diff
@@ -135,30 +161,52 @@ void testGetResourceRulesWhenItThrowsAnException() throws Exception {
 
-        CompletableFuture<AuditablePolicyRecordResponse> completableFuture = asyncProxy.getRecordRules(AUDITABLE_POLICY_RESOURCE_RESPONSE);
-        AuditablePolicyRecordResponse response = completableFuture.get();
-        assertThat(response.getPolicyResponse()).isNotNull();
-        assertThat(response.getAuditErrorMessage()).isNull();
+        // Then
+        assertAll("Check the values of the returned AuditablePolicyRecordResponse object",
+                () -> assertThat(response.getPolicyResponse())
+                        .as("Check the PolicyRequest value is not null")
+                        .isNotNull(),
+
+                () -> assertThat(response.getAuditErrorMessage())
+                        .as("Check the AuditErrorMessage value is null")
+                        .isNull()
+        );
     }
 
     /**
-     * Test for when Rules are not found for the record.  This should produce an {@link AuditablePolicyRecordResponse}
+     * Test for when Rules are not found for the record. This should produce an {@link AuditablePolicyRecordResponse}
      * with a {@code PolicyResponse} that has an empty {@code Rules} set and with {@code AuditErrorMessage}.
      *
      * @throws Exception if the test fails
      */
     @Test
     void testGetRecordRulesWithNoPolicyRecord() throws Exception {
+        // When
         when(hierarchyProxy.getRecordRules(any())).thenThrow(new NoSuchPolicyException("Test"));
-
-        CompletableFuture<AuditablePolicyRecordResponse> completableFuture = asyncProxy.getRecordRules(AUDITABLE_POLICY_RESOURCE_RESPONSE_WITH_NO_RULES);
-        AuditablePolicyRecordResponse response = completableFuture.get();
+        AuditablePolicyRecordResponse response = asyncProxy.getRecordRules(AUDITABLE_POLICY_RESOURCE_RESPONSE_WITH_NO_RULES).get();
         PolicyResponse policyResponse = response.getPolicyResponse();
-        assertThat(policyResponse).isNotNull();
-        assertThat(policyResponse.getRules()).isNotNull();
-        assertThat(policyResponse.getRules().containsRules());
-        assertThat(response.getAuditErrorMessage()).isNotNull();
+
+        // Then
+        assertAll(
+                () -> assertThat(policyResponse)
+                        .as("Check the PolicyRequest value is not null")
+                        .isNotNull(),
+
+                () -> assertThat(policyResponse.getRules())
+                        .as("Check the PolicyRequest contains a Rules object")
+                        .isNotNull(),
+
+                () -> assertThat(policyResponse.getRules().containsRules())
+                        .as("Check the Rules object does not contain any rules")
+                        .isEqualTo(Boolean.FALSE),
+
+                () -> assertThat(response.getAuditErrorMessage())
+                        .as("Check the AuditErrorMessage value is null")
+                        .isNotNull()
+        );
```

Comment:
```suggestion
        assertAll(
                () -> assertThat(policyResponse)
                        .as("Check the PolicyResponse and the variables inside the object are not null")
                        .usingRecursiveComparison()
                        .isNotNull(),

                () -> assertThat(policyResponse)
                        .as("Check the Rules inside the PolicyResponse are empty")
                        .extracting(PolicyResponse::getRules)
                        .isEqualTo(new Rules<>()),

                () -> assertThat(response.getAuditErrorMessage().getError())
                        .extracting(Throwable::getCause)
                        .as("Check the cause of the thrown error")
                        .isInstanceOf(NoSuchPolicyException.class)
                        .as("Check the message in the exception")
                        .extracting(Throwable::getMessage)
                        .isEqualTo("No rules found for the resource")
        );
```

Reply:
The assertion has been updated and is done slightly different to encompass a recursive comparison